components/shared/profile/mem.rs

```diff
@@ -266,3 +270,21 @@ pub enum ProfilerMsg {
     /// Triggers sending back the memory profiling metrics,
     Report(IpcSender<MemoryReportResult>),
 }
+
+thread_local!(static SEEN_POINTERS: LazyCell<RefCell<HashSet<*const c_void>>> = const {
+    LazyCell::new(|| RefCell::new(HashSet::new()))
```

Comment:
This could likely be a little simpler:

```suggestion
    LazyCell::new(Default::default)
```